src/js/providers/utils/quality-labels.ts

```diff
@@ -1,24 +1,36 @@
+import type { Level } from 'providers/data-normalizer';
+import type { GenericObject } from 'types/generic.type';
+
+type QualityLabels = {
+    [key: number]: string;
+};
+
 // Try and find a corresponding custom label. If there are no custom labels, create one using height, bandwidth, or both
-export function generateLabel(level, qualityLabels, redundant) {
+export function generateLabel(level: Partial<Level>, qualityLabels: QualityLabels | null, redundant: boolean): string {
     if (!level) {
         return '';
     }
     // flash provider uses bitrate instead of bandwidth
-    const bandwidth = level.bitrate || level.bandwidth;
+    const bandwidth = parseFloat(level.bitrate || level.bandwidth);
```

Comment:
We could end up with `NaN` here rather than `undefined`. OR it with 0 to ensure a finite number with falsyness:
`const bandwidth = parseFloat(level.bitrate || level.bandwidth) || 0;`